File: main_script.py
#!/usr/bin/env python
'''
	main_script.py
	---------------------------------------------------------
	This is the ASB tutorial master Python script, the main 
	entry point for running this tutorial in Python. You will 
	not need to modify this file.

	Usage:
		$ python main_script.py
'''

def runFile(num, args):
	'''Execute an additional python script
	
		- num {int} the script number
		- args {dict} Additional global variables that need
		  to be defined for this script to run. 
	'''
	fn = 'python_scripts/script_%d.py' % num
	f = open(fn)
	code = compile(f.read(), fn, 'exec')
	exec(code, args)
	f.close()
	return None


def main():
	'''Run the tutorial

	   This function also gracefully handles when users 
	   do not have required third-party packages.
	'''
	try:
		import pandas as pd
		import matplotlib.pyplot as plt
		fig, ax = plt.subplots(1, 1)
		ax.axis('off')
		for i in range(1,6):
			runFile(i, {
				'pd': pd, 
				'plt': plt, 
				'ax': ax, 
				'fig': fig})
		plt.show()
	except ImportError as err:
		mod = str(err).split(' ')[-1].replace("'", '')
		msg = '\nERROR: %s is not installed. ' % mod
		msg += 'Try running:\n\n\t$ conda install %s' % mod
		print(msg)

if __name__ == '__main__':
	main()

File: main_script.R
#!/usr/bin/env Rscript
# 
#   main_script.R
# 	---------------------------------------------------------
# 	This is the ASB tutorial master R script, the main 
# 	entry point for running this tutorial in R. You will 
# 	not need to modify this file.
# 
# 	Use "Source with Echo" to run this script in its 
#   entirety in RStudio.
#

# Install ggplot package if missing
if ('ggplot2' %in% rownames(installed.packages()) == F){
  install.packages('ggplot2')
} 

library(ggplot2)

# Import subscript objects
for (i in seq(1, 5)) {
  fn <- paste0('R_scripts/script_', i, '.R')
  source(fn)
}

canvas <- ggplot(NULL) + 
  theme(axis.title=element_blank(),
        axis.text=element_blank(),
        axis.ticks=element_blank(),
        panel.background=element_blank(),
        panel.grid=element_blank())

# Implement script_1.R
canvas <- drawPeach(canvas)

# Implement script_2.R
canvas <- drawLeaf(canvas)

# Implement script_3.R
canvas <- drawStemAndHalo(canvas)

# Implement script_4.R
canvas <- drawASBLogo(canvas)

# Implement script_5.R
canvas <- drawSubtitle(canvas)

canvas

